### DISCUSSION TITLE
How to check if a variable is not defined?

### DISCUSSION THREAD

**Discussion by ilg-ul** [mesonbuild/meson#12411] [Q&A] [answered]
For user friendliness reasons, I would like to check if a variable is not defined and issue a proper error message instead of the current `ERROR: Unknown variable`. 

Is this possible?

**Comment by eli-schwartz** [comment]
What is improper about the error message that meson produces? What error message do you want to give instead?

Note that `get_variable('varname', 'fallback value')` exists.

**Reply by ilg-ul** [reply]
> What error message do you want to give instead?

The project uses multiple configuration files, the definitions are expected to be in one file, written by the user, and the error is issued from a different one, included from something like a library. 

The similar parallel CMake configuration in the same project reads:

```
if(NOT DEFINED xpack_platform_compile_definition)
  message(FATAL_ERROR "Define xpack_platform_compile_definition in platform-native/cmake/dependencies.cmake")
endif()
```

> 'fallback value'

There are no fallback values.

**Reply by mitchgrout** [reply, marked as answer]
https://mesonbuild.com/Reference-manual_functions.html#is_variable

A one-to-one conversion from cmake to meson would be

```meson
if not is_variable('xpack_platform_compile_defintion')
  error('Define xpack_platform_compile_definition in <...>')
endif
```

however I would probably pick a more natural expression like

```meson
assert(is_variable('xpack_platfrm_compile_definition', ...), 'diagnostic message goes here')
```

---

There are other options on how to deal with this depending on how you're getting these values; if you could provide a simplified project structure, we could discuss how best to set this up?

**Reply by ilg-ul** [reply]
Thank you, @mitchgrout, that was helpful.
